packages/next/src/server/after/after.ts

```diff
@@ -0,0 +1,37 @@
+import { getExpectedRequestStore } from '../../client/components/request-async-storage.external'
+import { staticGenerationAsyncStorage } from '../../client/components/static-generation-async-storage.external'
+
+import { markCurrentScopeAsDynamic } from '../app-render/dynamic-rendering'
+import type { AfterTask } from './shared'
+
+/**
+ * This function allows you to schedule callbacks to be executed after the current request finishes.
+ */
+export function unstable_after<T>(task: AfterTask<T>) {
+  const callingExpression = 'unstable_after'
+
+  const requestStore = getExpectedRequestStore(callingExpression)
+
+  const { afterContext } = requestStore
+  if (!afterContext) {
+    throw new Error('Invariant: No afterContext in requestStore')
+  }
+  if (!afterContext.enabled) {
+    throw new Error(
+      'unstable_after() must be explicitly enabled by setting `experimental.after: true` in your next.config.js.'
+    )
+  }
+
+  const staticGenerationStore = staticGenerationAsyncStorage.getStore()
+
+  if (staticGenerationStore) {
+    if (staticGenerationStore.forceStatic) {
+      // When we are forcing static, after() is a no-op
```

Comment:
Honestly, i'm not sure about the approach of "log a warning in force-static" -- warnings are too easy to miss. I feel like if we really think using `after()` in a `force-static` route is a bug (because there's no default/empty behavior we can do, it's a side effect), then we should just bail out completely.